src/test/scala/pl/iterators/sealedmonad/SealedTests.scala

```diff
@@ -28,38 +32,67 @@ trait SealedTests[F[_]] extends Laws with SealedTestInstances {
+      "result flatMap short-circuits"                                                   -> forAll(laws.resultFlatMapElimination[A, ADT] _),
+      "semiflatMap consistent with flatMap"                                             -> forAll(laws.valueSemiflatMapReduction[A, B] _),
+      "result semiflatMap short-circuits"                                               -> forAll(laws.resultSemiflatMapElimination[A, ADT] _),
+      "complete consistent with result + flatMap"                                       -> forAll(laws.valueCompleteIdentity[A, ADT] _),
+      "completeWith consistent with complete + unit"                                    -> forAll(laws.completeWithCoherence[A, ADT] _),
+      "result complete short-circuits"                                                  -> forAll(laws.resultCompleteElimination[A, ADT] _),
+      "rethrow right does not change value"                                             -> forAll(laws.rethrowRightIdentity[A, ADT] _),
+      "rethrow left consistent with complete"                                           -> forAll(laws.rethrowLeftIdentity[ADT] _),
+      "attempt right consistent with map"                                               -> forAll(laws.attemptRightIdentity[A, B, ADT] _),
+      "attempt left consistent with complete"                                           -> forAll(laws.attemptLeftIdentity[A, ADT] _),
+      "attemptF consistent with attempt + unit"                                         -> forAll(laws.attemptFCoherence[A, B, ADT] _),
+      "ensure true identity"                                                            -> forAll(laws.ensureTrueIdentity[A, ADT] _),
+      "ensure false consistent with complete"                                           -> forAll(laws.ensureFalseIdentity[A, ADT] _),
+      "ensure consistent with ensureNot"                                                -> forAll(laws.ensureCoherence[A, ADT] _),
+      "ensure consistent with rethrow"                                                  -> forAll(laws.ensureRethrowCoherence[A, ADT] _),
+      "either identity"                                                                 -> forAll(laws.eitherIdentity[A, ADT] _),
+      "foldM consistent with flatMap"                                                   -> forAll(laws.foldMCoherentWithFlatMap[A, ADT] _),
+      "inspect does not change instance"                                                -> forAll(laws.inspectElimination[A, B, ADT] _),
+      "valueOr"                                                                         -> forAll(laws.valueOrIdentity[A, ADT] _),
+      "handleError"                                                                     -> forAll(laws.handleErrorIdentity[A, B, ADT] _),
+      "semiflatMap stack-safety"                                                        -> lzy(laws.semiflatMapStackSafety),
+      "inspect side effect not invoked"                                                 -> sideEffectsInspect.simplyInvoke,
```

Comment:
I think these tests should not be a part of ScalaCheck suite - they get converted to properties because they return boolean but nonetheless they don't take any randomized input.

Reply:
I think it's much butter to have them in SealedSpecs